freyja/sample_deconv.py

```diff
@@ -25,41 +25,32 @@ def buildLineageMap():
 	return mapDict
 
 
-def build_mix_and_depth_arrays(dirName,depthDir,muts):
-	df_mix = pd.DataFrame()
-	df_depths = pd.DataFrame()
-	fnames = os.listdir(dirName)
-
-	filetype =fnames[0].split('.')[-1]
-	for fn in tqdm(os.listdir(dirName)):
-		df = pd.read_csv(dirName+fn,sep='\t')
-		try:
-			df_depth  = pd.read_csv(depthDir+fn.split(filetype)[0]+'depth',sep='\t',header=None,index_col=1)
-		except:
-			continue
-		df['mutName'] = df['REF'] + df['POS'].astype(str) + df['ALT']### this only works for substitutions, but that's what we get from the usher tree
-		df = df.drop_duplicates(subset='mutName')
-		df.set_index('mutName',inplace=True)
-		keptInds = [dfi for jdi, dfi in  enumerate(df.index) if dfi  in muts]#meh... works but should be improved
-		df_mix = df_mix.append(pd.Series({kI:df.loc[kI,'ALT_FREQ'].astype(float) for kI in keptInds},name=fn))
-		df_depths = df_depths.append(pd.Series({kI:df_depth.loc[int(re.findall(r'\d+',kI)[0]),3].astype(float) for kI in muts},name=fn))
-	return df_mix,df_depths
-
-
-def reindex_dfs(df_barcodes,df_mix,df_depths):
+def build_mix_and_depth_arrays(fn, depthFn,muts):
+	df = pd.read_csv(fn,sep='\t')
+	df_depth  = pd.read_csv(depthFn,sep='\t',header=None,index_col=1)
+	df['mutName'] = df['REF'] + df['POS'].astype(str) + df['ALT']### this only works for substitutions, but that's what we get from the usher tree
+	df = df.drop_duplicates(subset='mutName')
+	df.set_index('mutName',inplace=True)
+	keptInds = [dfi for jdi, dfi in  enumerate(df.index) if dfi  in muts]#meh... works but should be improved
+	mix = pd.Series({kI:df.loc[kI,'ALT_FREQ'].astype(float) for kI in keptInds},name=fn)
+	depths = pd.Series({kI:df_depth.loc[int(re.findall(r'\d+',kI)[0]),3].astype(float) for kI in muts},name=fn)
```

Comment:
I'm not entirely sure I know what this is doing. It look a `Series` is being created, from `df_depth`, using all entries in `kI` which are composed of one or more digits for the index, and the column named `3`?

Reply:
Yes, this is taking all rows of the depth file that correspond to lineage defining mutations, and grabbing the column that has the corresponding sequencing depth info (mpileup third column — could also use iloc here)